x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_logic.test.ts

```diff
@@ -170,67 +214,195 @@ describe('EngineLogic', () => {
         expect(EngineLogic.actions.setEngineData).toHaveBeenCalledWith(mockEngineData);
       });
 
-      it('handles errors', async () => {
+      it('handles 4xx errors', async () => {
         mount();
         jest.spyOn(EngineLogic.actions, 'setEngineNotFound');
-        http.get.mockReturnValue(Promise.reject('An error occured'));
+        http.get.mockReturnValue(Promise.reject({ response: { status: 404 } }));
 
         EngineLogic.actions.initializeEngine();
         await nextTick();
 
         expect(EngineLogic.actions.setEngineNotFound).toHaveBeenCalledWith(true);
       });
+
+      it('handles 5xx errors', async () => {
+        mount();
+        jest.spyOn(EngineLogic.actions, 'setEngineNotFound');
+        http.get.mockReturnValue(Promise.reject('An error occured'));
+
+        EngineLogic.actions.initializeEngine();
+        await nextTick();
+
+        expect(flashErrorToast).toHaveBeenCalledWith('Could not fetch engine data', {
+          text: expect.stringContaining('Please check your connection'),
+          toastLifeTimeMs: 3750,
+        });
+      });
+    });
+
+    describe('pollEmptyEngine', () => {
+      beforeEach(() => jest.useFakeTimers());
```

Comment:
If you don't call `useRealTimers` to clean this up, will this bleed into subsequent tests where you *don't* intend to use fake timers?

Reply:
No other tests/fns in this file use timers, so I'm not super worried about it. TBH, I also can't recall any scenarios where I've actually wanted real timers in a unit tests 🤔 

TBH, this group of tests did give me a decent amount of grief when unit testing. I originally was declaring just a flat `jest.useFakeTimers();` at the start of the describe block and not in a beforeEach, and it was causing tests to fail when run as a group but pass when run with `it.only()`. Per the [Jest mock timers docs](https://jestjs.io/docs/timer-mocks),

> While you can call jest.useFakeTimers() or jest.useRealTimers() from anywhere (top level, inside an it block, etc.), it is a global operation and will affect other tests within the same file. Additionally, you need to call jest.useFakeTimers() to reset internal counters before each test. If you plan to not use fake timers in all your tests, you will want to clean up manually, as otherwise the faked timers will leak across tests:

So `useFakeTimers` in a beforeEach() worked great, but for some reason(?!) I also needed `jest.clearAllTimers()` in an after each for tests to pass. I tried `afterEach(() => jest.useRealTimers())` to try and reset state as well but for some reason that _still_ caused certain tests to fail. Honestly I was banging my head against this a good amount yesterday so at some point I was like "f it, tests are passing, gotta move on".

Haha sorry, ok, that fun story aside - yes, you know what, that's a good point, I'm going to go ahead and add an `afterAll(() => jest.useRealTimers())` to this block

Reply:
https://github.com/elastic/kibana/pull/103041/commits/308a8804329fd8beb1a8d63c7570d1d42110a685